src/controller/student.controller.js

```diff
@@ -0,0 +1,131 @@
+import Student from '../model/Student';
+import responseHandler from '../response/response.handler';
+import enums from './controller.enums';
+
+export async function createStudent(req, res) {
+  if (req.user && req.user.role === enums.role.ADMIN) {
+    let studentData = new Student(req.body);
+    const student = new Student(studentData);
+    await student
+      .save()
+      .then((data) => {
+        res.status(200).json(data);
+      })
+      .catch((error) => {
+        res.status(500).json(error.message);
+      });
+  } else {
+    return responseHandler.respond(res, enums.roleIssue.ONLY_ADMIN);
+  }
+}
+
+export async function getAllStudents(req, res) {
+  await Student.find({})
+    .then((students) => {
+      res.status(200).json(students);
+    })
+    .catch((error) => {
+      res.status(500).json(error.message);
+    });
+}
+
+export async function getStudentByID(req, res, next) {
+  if (req.params && req.params.id) {
+    await Student.findById(req.params.id)
+      .populate({
+        path: 'student',
+        populate: {
+          path: 'student',
+          model: 'users',
+          select:
+            '_id firstname lastname dateofbirth address1 address2 city province grade imageurl achievements parent phone email username',
```

Comment:
In enum
'STUDENT_SELECT': '_id firstname lastname dateofbirth address1 address2 city province grade imageurl achievements parent phone email username'

In controller
enums.student.STUDENT_SLECT

**Look question.controller.js file**